dist/tools/pyterm/pyterm

```diff
@@ -653,6 +657,10 @@ class SerCmd(cmd.Cmd):
             if not ignored:
                 self.process_line(line)
 
+        if self.line_regex and self.line_regex.match(line):
+            self.do_PYTERM_exit("")
+            os._exit(0)
```

Comment:
This is a bit ugly, but otherwise one would need to terminate the cmdloop thread somehow since this code is not executed on the main thread.